packages/reactive-element/src/reactive-element.ts

```diff
@@ -1069,9 +1069,11 @@ export abstract class ReactiveElement
       this.constructor as typeof ReactiveElement
     ).__attributeNameForProperty(name, options);
     if (attr !== undefined && options.reflect === true) {
+      const converter = options.converter;
       const toAttribute =
-        (options.converter as ComplexAttributeConverter)?.toAttribute ??
-        defaultConverter.toAttribute;
+        (converter as ComplexAttributeConverter)?.toAttribute?.bind(
+          converter
+        ) ?? defaultConverter.toAttribute;
```

Comment:
If it wouldn't be a breaking change, it'd be nice and simple to just do:

```ts
const attrValue = (converter ?? defaultConverter).toAttribute?.(value, options.type);
```

but this would change the behavior of the case where there's a converter but with no `toAttribute`. cc @sorvell

Reply:
It would be a breaking change in case a converter is provided as function. Quoting from docs:

> If converter is a **function**, it is used in place of **fromAttribute**